src/vs/workbench/services/dialogs/browser/abstractFileDialogService.ts

```diff
@@ -86,12 +86,24 @@ export abstract class AbstractFileDialogService implements IFileDialogService {
 		}
 
 		if (!candidate) {
-			return this.pathService.userHome({ preferLocal: schemeFilter === Schemas.file });
+			return this.preferredHome(schemeFilter);
 		}
 
 		return resources.dirname(candidate);
 	}
 
+	async preferredHome(schemeFilter = this.getSchemeFilterForWindow()): Promise<URI> {
+		const dialogHomePath = this.configurationService.getValue<string>('files.dialog.homePath');
+		const userHomePromise = this.pathService.userHome({ preferLocal: schemeFilter === Schemas.file });
+		if (!dialogHomePath) {
+			return userHomePromise;
+		}
+
+		// Normalize homePath setting so simple file dialog works correctly
+		const path = (await this.pathService.fileURI(dialogHomePath)).path;
```

Comment:
I am sorry, but this change looks like a hack. It cannot go in like that.

Reply:
As I understand things, `preferredHome` must return the type (i.e. scheme) of URI that the caller requested (e.g. 'file' or 'vscode-remote'). And when I'm running through the debugger the `this.pathService.fileUri` method always returned a 'file://' URI regardless of whether I worked locally or remotely (using TestResolver).

I will push some more inline comments, plus a change to use `inspect` to retrieve the setting value because when working on a remote I don't think we should fall back to a local user-level path value if no remote one exists. @alexr00 might have a view about this.

Reply:
Lets step back. You have a configured path that is not a URI and you want to convert that to the best URI for the environment you are in. That should only be 1 call and not 2. I suggest to use this:

https://github.com/microsoft/vscode/blob/ff3369f2329fb18601dcd1f51655cd5445535974/src/vs/base/common/resources.ts#L433

With the `pathService.defaultUriScheme`, e.g. like here:

https://github.com/microsoft/vscode/blob/ff3369f2329fb18601dcd1f51655cd5445535974/src/vs/workbench/services/textfile/browser/textFileService.ts#L587

This should make sure that the non-URI path gets converted to the best local URI matching the environment.

Reply:
Genius! I have pushed the change.